overlord/devicestate/systems.go

```diff
@@ -237,11 +237,28 @@ type setupInfoGetter struct {
 
 func (ig *setupInfoGetter) ComponentInfo(st *state.State, cref naming.ComponentRef, snapInfo *snap.Info) (info *snap.ComponentInfo, path string, present bool, err error) {
 	// components will come from one of these places:
+	//   * passed into the task via a list of side infos (these would have
+	//     come from a user posting snaps via the API)
 	//   * have just been downloaded by a task in setup.ComponentSetupTasks
 	//   * already installed on the system
-	//
-	// TODO:COMPS: handle components passed into the task via a list of side
-	// infos (these would have come from a user posting components via the API)
+
+	for _, l := range ig.setup.LocalComponents {
```

Comment:
There is a line below:
`logger.Debugf("requested info for component %q being installed during remodel", cref)`
that would need to be moved right before this for.